modules/drivers/sqlite/src/metabase/driver/sqlite.clj

```diff
@@ -123,9 +124,7 @@
 
 ;; See also the [SQLite Date and Time Functions Reference](http://www.sqlite.org/lang_datefunc.html).
 
-(defmethod sql.qp/date [:sqlite :default]
-  [driver _ expr]
-  (sql.qp/->honeysql driver expr))
```

Comment:
@camsaul This change fixes the field type checking for SQLite. Please double-check this fix is reasonable before I merge.

This method is called when `->honeysql` is called on a field. 
See: 
https://github.com/metabase/metabase/blob/2eb0f03a10efd5a4a77260a85e3fa0ee06fdcf6b/src/metabase/driver/sql/query_processor.clj#L491
And:
https://github.com/metabase/metabase/blob/2eb0f03a10efd5a4a77260a85e3fa0ee06fdcf6b/src/metabase/driver/sql/query_processor.clj#L417-L421

The method body reduces to the equivalent of `(hx/unwrap-typed-honeysql-form expr)`, which removes the type info from the typed identifier form.